src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -925,12 +942,13 @@
           [value id])))
 
 (defn realize-records!
-  "Inserts the records (maps) into the named database and returns them
+  "Inserts the records (maps) into the named table and returns them
   with their new :id values."
-  [database records]
-  (map #(assoc %2 :id %1)
-       (map :id (apply sql/insert-records database records))
-       records))
+  [table records]
+  (when (seq records)
```

Comment:
Yes - I believe insert! chokes when given no arguments, and insert-records didn't.